### PR TITLE
boards: opentitan: Fixup existing test cases

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

Fixup the current test cases and update the documentation to be simpler to follow

### Testing Strategy

`make test-hardware`

### TODO or Help Wanted

N/A

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: boards/opentitan/README.md [hunk 1]
```diff
@@ -28,24 +28,24 @@ or simulator built from that version of the OpenTitan codebase.
 Programming
 -----------
 
-The latest supported version (commit SHA) of OpenTitan is specified in the [OPENTITAN_SUPPORTED_SHA](https://github.com/tock/tock/blob/master/boards/opentitan/earlgrey-cw310/Makefile) make variable found:
->  boards/opentitan/earlgrey-cw310/Makefile
-
-In *general* it is recommended that users start with the commit specified by `OPENTITAN_SUPPORTED_SHA` as newer
+It is recommended that users start with the commit specified by `Earlgrey-M2.5.2-RC0` as newer
 versions **have not been** tested.
```
Comment (bradjc) [inline boards/opentitan/README.md:32]: We're still using a SHA from https://github.com/tock/tock/pull/3639

Comment (alistair23) [reply boards/opentitan/README.md:32]: Who is we? This is what the current Makefile has:

```shell
# OpenTitan commit 21ce4e9761abdf5c919b46e5ae64a5a8e24992f7 is Earlgrey-M2.5.2-RC0.
# This commit is what is being taped out in the first Earlgrey chip.
OPENTITAN_SUPPORTED_SHA := 21ce4e9761abdf5c919b46e5ae64a5a8e24992f7
```

Comment (bradjc) [reply boards/opentitan/README.md:32]: Tock, we are using the taped out version but we will still need to specify a SHA hash.

**Comment by twilfredo** [comment]
is the consolidation of the flash/tickv tests into `mod.rs` to preserve order?

File: boards/opentitan/README.md (hunk 1 above)
Comment (alistair23) [reply boards/opentitan/README.md:32]: Yeah, we are using the tag from `Earlgrey-M2.5.2-RC0`. The specific hash is specified in the Makefile

**Comment by alistair23** [comment]
> is the consolidation of the flash/tickv tests into `mod.rs` to preserve order?

Yes

File: boards/opentitan/README.md (hunk 1 above)
Comment (bradjc) [reply boards/opentitan/README.md:32]: No I'm saying we can't use that tag we need to use this commit: https://github.com/tock/tock/pull/3639#issuecomment-1724140257

Comment (alistair23) [reply boards/opentitan/README.md:32]: I'm not changing the tag here. It's the same one we are already using. If there is a new one to use we can update to that

Comment (alistair23) [reply boards/opentitan/README.md:32]: Awe! Is your point that we shouldn't use a release tag and instead should continue to use SHAs?

Comment (bradjc) [reply boards/opentitan/README.md:32]: I think the relevant discussion is in that PR, but the gist is that timing for FPGA can still change without changing the taped out HW.

Comment (alistair23) [reply boards/opentitan/README.md:32]: I feel like that should be discussed in that PR. Currently we are using a release tag, so I think it makes sense to say that in the README. If we decided to change we can just update the README

Comment (bradjc) [reply boards/opentitan/README.md:32]: I don't want to merge this PR until after that one, so we can just wait. I was just hinting to preemptively remove the conflict.

Comment (alistair23) [reply boards/opentitan/README.md:32]: Why wait on https://github.com/tock/tock/pull/3639? It doesn't even apply and this PR is ready and fixes broken test cases

**Review by twilfredo — APPROVED** [review]

File: boards/opentitan/README.md (hunk 1 above)
Comment (bradjc) [reply boards/opentitan/README.md:32]: Backout the readme changes so this just updates the tests? That makes this an easy merge.

Comment (alistair23) [reply boards/opentitan/README.md]: Done!

**Review by bradjc — APPROVED** [review]

**Pull request merged**